src/sgb-xi/codes.ts

```diff
@@ -0,0 +1,289 @@
+}
+export type UmsatzsteuerBefreiungSchluessel = keyof typeof umsatzsteuerBefreiungSchluessel;
+
+// 2.14.2 Schlüssel Zuschlagsart
+export const zuschlagsartSchluessel = {
+    "1": "ambulant",
+    "2": "stationär (teil-/vollstationär)",
+    "3": "Kurzzeitpflege",
+}
+export type ZuschlagsartSchluessel = keyof typeof zuschlagsartSchluessel;
+
+// 2.14.3 Schlüssel Zuschlag
+export const zuschlagSchluessel = {
+    "00": "wenn kein Schlüssel vorhanden, (dann aber Klartext angeben)",
+    "02": "Prozentuale anteilige Wegegebühren wegen gleichzeitiger LE SGB V und SGB XI",
+    "03": "Anteilige Wegegebühren bei betreutes Wohnen",
+    "04": "Höhere Vergütung für Einsatz einer zweiten Pflegekraft ",
+    "05": "Zu-/Abwahl von einzelnen Leistungen",
+    "06": "2 Personen",
+    "07": "Demente",
+    "08": "Apalliker",
+    "09": "Härtefälle",
+    "10": "Taxi",
+    "11": "Abwesenheit bei Urlaub (gilt für stationär)",
+    "12": "Abwesenheit bei Krankheit (gilt für stationär)",
+    "13": "Abwesenheit bei Kur (gilt für stationär)",
+    "14": "Abwesenheit bei Reha (gilt für stationär)",
+    "15": "Abwesenheit bei Wochenendurlaub (gilt für stationär)",
+    "16": "Sonstige Abwesenheit",
+    "17": "Abwesenheit bei Krankenhausaufenthalt",
+    "18": "Ausbildungsumlage nach PflAFinV",
+    "19": "MRE-Zuschlag",
+    "21": "Ungünstige Zeiten",
+    "22": "Ungünstige Tage",
+}
+export type ZuschlagSchluessel = keyof typeof zuschlagSchluessel;
+
+// n.a. Schlüssel Zuschlagszuordnung
+export const zuschlagszuordnungSchluessel = {
+    "1": "Leistung",
+    "2": "Wegegebühr",
+}
+export type ZuschlagszuordnungSchluessel = keyof typeof zuschlagszuordnungSchluessel;
+
+// 2.15 Schlüssel Pflegegrad
+export const pflegegradSchluessel = {
+    "1": "Pflegegrad 1",
+    "2": "Pflegegrad 2",
+    "3": "Pflegegrad 3",
+    "4": "Pflegegrad 4",
+    "5": "Pflegegrad 5",
+}
+export type PflegegradSchluessel = keyof typeof pflegegradSchluessel;
+
+// 2.16 Schlüssel Zuschlagsberechnung
+export const zuschlagsberechnungSchluessel = {
+    "01": "Punktzahl absolut",
+    "02": "Prozentsatz zur Punktzahl",
+    "03": "Punktwert absolut",
+    "04": "Prozentsatz zum Punktwert",
+    "05": "Basiswert + Addition (Subtraktion) Betrag absolut bzw. Ersatzwert",
+    "06": "Basiswert + Addition (Subtraktion) Punktzahl absolut bzw. Ersatzwert",
+    "07": "Basiswert + Addition (Subtraktion) Punktwert absolut bzw. Ersatzwert",
+    "08": "monatlicher Betrag",
+    "09": "Ersatzwert mtl. Punktzahl",
+    "10": "Ersatzwert mtl. Punktwert",
+    "11": "Prozentsatz zum Betrag",
+    "12": "Betrag absolut",
+    "13": "Prozentsatz zum Betrag-Ersatzwert aller ZUS",
+    "14": "täglicher Betrag (absoluter Betrag * Anzahl / Menge aus Segment ELS)",
+    "15": "Prozentsatz zum Punktzahl-Ersatzwert aller ZUS",
+    "16": "Prozentsatz zum Punktwert-Ersatzwert alles ZUS",
+    "17": "Basiswert + Prozentsatz absolut bzw. Ersatzwert",
```

Comment:
This is probably not relevant for writing, but it would be for parsing, so maybe it makes sense to be consistent here:

Did you check if the key value is documented to be a number or a text? This makes a difference because if it is a number, the sender may either send for example a "01" or a "1".
At least for parsing this makes a difference, because if either of those two could arrive, then `zuschlagsberechnungSchluessel[parsedValue]` won't work.

Reply:
All the keys are strings. Always.

Reply:
What about those that are not?

For example, on page 15 in https://gkv-datenaustausch.de/media/dokumente/leistungserbringer_1/pflege/technische_anlagen_aktuell_2/TA1_ANH5_20170907_105_oA.pdf

, the "Art der Datenlieferung" is a number. So, this field could for example be sent as "7" or "07".

This is not all theoretic, by chance I found another place where depending on the health insurance group, data was sent differently:

- AOK sends `UNH+00001+KOTR:01'`
- Knappschaft sends `UNH+1+KOTR:2'`

Reply:
Ok, I see this is an issue for parsing. 
I suggest that you parse it as number first: `parseInt("9")` has always the same result, no matter how many Zeros are padded at the start of the string. Then convert it back to a string, so we can handle all keys internally as strings in a consistent way. Would that solve the problem of parsing data with inconsistent Schlüssel formats?

Reply:
That would be a solution, so the Schlüssel would always be a string without leading zeroes right?
Though, does anything speak against something like this?
```typescript
export const datenlieferungsArtSchluessel = {
    7: "digitalisierte Rechnungs- und Abrechnungsdaten (PLGA und PLLA)",
    21: "Rechnung Papier",
    24: "maschinenlesbare Belege",
    26: "Verordnung",
    28: "Urbelege zu einer digitalen Abrechnung",
    29: "maschinenlesbare Belege einschließlich der dazugehörigen Urbelege"
}
```
?

Reply:
Read up on that.

> Property names must be strings. This means that non-string objects cannot be used as keys in the object. Any non-string object, including a number, is typecasted into a string via the toString method.

So, it is possible to write it like that, but in the end it will be a string in any case. However, Typescript type inference does this:

![image](https://user-images.githubusercontent.com/4661658/114775267-6964a180-9d71-11eb-9304-072b3c479813.png)

So, which do you prefer?